Compiling 20200314 tag fails if tile path is enabled

Hello all. I maintain the FreeBSD port of Fontforge and I was attempting to update it to the 20200314 tag. However, I ran into a compilation error when I had enabled the tile path option in the CMake configuration. The full build log can be found here:

https://poudriere.cyberbotx.com:8766/data/local-default/2020-03-19_09h39m51s/logs/errors/fontforge-20200314.log

The relevant bits:
```
[00:03:00] [282/314] /usr/bin/cc  -I/wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforge -I/wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/inc -Iinc -isystem /usr/local/include/gtk-3.0 -isystem /usr/local/include/pango-1.0 -isystem /usr/local/include/glib-2.0 -isystem /usr/local/lib/glib-2.0/include -isystem /usr/local/include -isystem /usr/local/include/fribidi -isystem /usr/local/include/cairo -isystem /usr/local/include/pixman-1 -isystem /usr/local/include/freetype2 -isystem /usr/local/include/uuid -isystem /usr/local/include/libdrm -isystem /usr/local/include/libpng16 -isystem /usr/local/include/harfbuzz -isystem /usr/local/include/gdk-pixbuf-2.0 -isystem /usr/local/include/gio-unix-2.0 -isystem /usr/local/include/libepoll-shim -isystem /usr/local/lib/libffi-3.2.1/include -isystem /usr/local/include/libxml2 -isystem /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/freetype/include -isystem /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/freetype/include/freetype -isystem /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/freetype/src/truetype -isystem /usr/local/include/readline -isystem /usr/local/include/python3.7m -O2 -pipe  -fstack-protector-strong -fno-strict-aliasing -O2 -pipe  -fstack-protector-strong -fno-strict-aliasing   -Werror=implicit-function-declaration -Werror=int-conversion -fdiagnostics-color=always -pthread -D_THREAD_SAFE -std=gnu99 -MD -MT fontforgeexe/CMakeFiles/fontforgeexe.dir/tilepath.c.o -MF fontforgeexe/CMakeFiles/fontforgeexe.dir/tilepath.c.o.d -o fontforgeexe/CMakeFiles/fontforgeexe.dir/tilepath.c.o   -c /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c
[00:03:00] FAILED: fontforgeexe/CMakeFiles/fontforgeexe.dir/tilepath.c.o 
[00:03:00] /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c:574:21: error: no member named 'y' in 'struct fitpoint'
[00:03:00]                 to->p.x = x1; to->y = y1;
[00:03:00]                               ~~  ^
[00:03:00] /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c:613:17: error: use of undeclared identifier 'tp'
[00:03:00]     new->me.x = tp.p.x; new->me.y = tp.p.y;
[00:03:00]                 ^
[00:03:00] /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c:613:37: error: use of undeclared identifier 'tp'
[00:03:00]     new->me.x = tp.p.x; new->me.y = tp.p.y;
[00:03:00]                                     ^
[00:03:00] /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c:849:14: warning: 20 enumeration values not handled in switch: 'et_noevent', 'et_charup', 'et_mousemove'... [-Wswitch]
[00:03:00]     switch ( event->type ) {
[00:03:00]              ^
[00:03:00] /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c:869:14: warning: 17 enumeration values not handled in switch: 'et_noevent', 'et_charup', 'et_mousemove'... [-Wswitch]
[00:03:00]     switch ( event->type ) {
[00:03:00]              ^
[00:03:00] /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c:1510:14: warning: 20 enumeration values not handled in switch: 'et_noevent', 'et_charup', 'et_mousemove'... [-Wswitch]
[00:03:00]     switch ( event->type ) {
[00:03:00]              ^
[00:03:00] /wrkdirs/usr/ports/print/fontforge/work/fontforge-20200314/fontforgeexe/tilepath.c:1529:14: warning: 17 enumeration values not handled in switch: 'et_noevent', 'et_charup', 'et_mousemove'... [-Wswitch]
[00:03:00]     switch ( event->type ) {
[00:03:00]              ^
[00:03:00] 4 warnings and 3 errors generated.
```
If I had to guess on the fixes:
* `to->y = y1;` needs to be `to->p.y = y1;`
* `new->me.x = tp.p.x; new->me.y = tp.p.y;` needs to be `new->me.x = fp.p.x; new->me.y = fp.p.y;`

I'm not sure if those are what is actually needed to make this compile or not, though.

--- a/fontforgeexe/tilepath.c
+++ b/fontforgeexe/tilepath.c
@@ -571,7 +571,7 @@ static void AdjustPoint(TD *td,Spline *spline,bigreal t, FitPoint *to) {
 	} else {
 	    bigreal s=(dy1*dx2/dy2-dx1);
 	    if ( s>-.00001 && s<.00001 ) {	/* essentially parallel */
-		to->p.x = x1; to->y = y1;
+		to->p.x = x1; to->p.y = y1;
 	    } else {
 		bigreal t1 = (x1-x2- dx2/dy2*(y1-y2))/s;
 		to->p.x = x1 + dx1*t1;
@@ -610,7 +610,7 @@ static SplinePoint *TDMakePoint(TD *td,Spline *old,real t) {
 
     AdjustPoint(td,old,t,&fp);
     new = chunkalloc(sizeof(SplinePoint));
-    new->me.x = tp.p.x; new->me.y = tp.p.y;
+    new->me.x = fp.p.x; new->me.y = fp.p.y;
     new->nextcp = new->me;
     new->prevcp = new->me;
     new->nonextcp = new->noprevcp = true;

